main/src/library/Iterable.flix

```diff
@@ -17,17 +17,26 @@
 ///
 /// A trait for immutable data structures that can be iterated.
 ///
-pub trait Iterable[t: Type -> Type] {
+pub trait Iterable[t] {
+    ///
+    /// The element type of the Iterable.
+    ///
+    type Elm[t]: Type
+
+    ///
+    /// The associated effect of the Iterable.
+    ///
+    type Aef[t]: Eff
```

Comment:
We have the ability to specify a default effect:

```
    type Aef[t]: Eff = Pure
```

then instances, if they are pure, can simply omit a definition of `Aef`.

Reply:
I've added the default and removed `Aef` definitions when it was Pure.